app/api/catalogues/[catalogueId]/quiz/submit/route.ts

```
@@ -0,0 +1,71 @@
+ * Upon successful validation, it retrieves the correct answers from the database using the provided catalogue
+ * identifier, calculates the quiz score, and returns the result as a JSON response.
+ *
+ * If the payload fails validation, it returns a 400 status with the validation errors. Any other errors result
+ * in a 500 status with a generic error message.
+ *
+ * @param req - The HTTP request containing the quiz submission data.
+ * @param params - An object resolving to route parameters including the catalogue identifier.
+ * @returns A JSON response with the quiz score or an error message.
+ */
+
+export async function POST(req: Request, { params }: { params: Promise<any> }) {
```

ESLint

[error] Unexpected any. Specify a different type.

(@typescript-eslint/no-explicit-any)